_posts/2016-06-17-using-xhprof.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: "Using XHProf to profile your Drupal module"
+date: 2016-06-17
+author: Tim Stallmann
+tags: php performance drupal drupal-planet module-development
+summary: A case study of using XHProf to profile a Drupal module.
+drupal_planet_summary:
+  Second part in a series of how to use XHProf effectively within a VM for a Drupal website.
+---
+
+After you've [installed XHProf](/2016/05/26/installing-xhprof.html), what's next? How can you make use of its recommendations to tune a module or a website? Unfortunately there are no hard-and-fast rules for optimizing code to run more efficiently. And I'm no expert on the matter, either. What I can offer here is my own experience trying to optimize a D8 module using XHProf.
+
+### Understanding an XHProf run report
+
+The XHProf GUI displays the result of a given profiler run, or a group of runs. It can even compare 2 runs, but we'll get to that in a minute. If you followed my previous post, you should have the `xhprof_html` directory symlinked into the root web directory for your Drupal site; so visiting `<my-local-site>/xhprof/` should give you a list of all available stored run IDs, and you can click through one of those to view a specific run report.
+
+You can also go directly to a specific run report via the URL `<my-local-site>/xhprof/index.php?run=<run-id>&source=<source-id>` (which you should have been logging already via an `echo` statement or `dblog` if you followed the last post).
+
+The core of the run report is a table of each function or method which your code called while it was being profiled, along with a set of statistics about that function. This allows you to understand which parts of your code are most resource-intensive, and which are being called frequently in the use case that you're profiling. Clicking on any one of the column headers will sort the list by that metric. To understand this report, it's helpful to have some terminology:
+
```

Comment:
Thoughts on including a screenshot here? I know there's one included below, but if I were just reviewing this article before I did anything myself I'd benefit from seeing the table you're talking about, even if it's the exact same screenshot you use later on.